synapse/http/site.py

```diff
@@ -96,25 +96,34 @@ def __repr__(self):
     def get_request_id(self):
         return "%s-%i" % (self.get_method(), self.request_seq)
 
-    def get_redacted_uri(self):
-        uri = self.uri
+    def get_redacted_uri(self) -> str:
+        """Gets the redacted URI associated with the request (or placeholder if not
+        method has yet been received).
+
+        Note: This is necessary as the placeholder value in twisted is str
+        rather than bytes, so we need to sanitise `self.uri`.
+
+        Returns:
+            The redacted URI as as string.
+        """
+        uri = self.uri  # type: Union[bytes, str]
         if isinstance(uri, bytes):
-            uri = self.uri.decode("ascii", errors="replace")
+            uri = uri.decode("ascii", errors="replace")
         return redact_uri(uri)
 
-    def get_method(self):
+    def get_method(self) -> str:
         """Gets the method associated with the request (or placeholder if not
         method has yet been received).
```

Comment:
```suggestion
        """Gets the method associated with the request (or placeholder if method
        has not yet been received).
```

Oh, it was a c/p. Well same here too then.